src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -433,7 +428,7 @@ private function resolveCommands(string $value, array $loadedVars): string
             )
         /x';
 
-        return preg_replace_callback($regex, function ($matches) use ($loadedVars) {
+        return preg_replace_callback($regex, function ($matches) use ($loadedVars): ?string {
```

Comment:
return type is not needed, this closure is not exposed

Reply:
If i correctly understand the comment, Symfony consider that only the public API could be stricter? 

Is there any internal design considerations that prevent the "non-exposed" calls to be stricter? Performances? Developer experience? Future evolutions / rewriting? 🤔